src/test/relation_engine_client/test_integration.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+import os
+from uuid import uuid4
+
+from src.relation_engine_client import REClient
+from src.relation_engine_client.exceptions import RERequestError, RENotFound
+
+_API_URL = os.environ.get('RE_API_URL', 'http://localhost:5000')
+_VERT_COLL = 'test_vertex'
+_EDGE_COLL = 'test_edge'
+_TOK_ADMIN = 'admin_token'
+_TOK_USER = 'non_admin_token'
+_TOK_INVALID = 'invalid_token'
+
+
+class TestREClientIntegration(unittest.TestCase):
+    """Integration tests for the REClient package."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.client = REClient(_API_URL, _TOK_ADMIN)
+        pass
+
+    def test_admin_query_ok(self):
+        _id = self._save_test_vert()
+        bind_vars = {'id': _id}
+        query = f"FOR vert IN {_VERT_COLL} FILTER vert._key == @id RETURN vert"
+        result = self.client.admin_query(query, bind_vars)
+        self.assertEqual(result['count'], 1)
+        self.assertEqual(result['results'][0]['_key'], _id)
+
+    def test_admin_empty_query(self):
+        bind_vars = {'id': 'xyz'}
+        with self.assertRaises(RERequestError) as ctx:
+            self.client.admin_query("", bind_vars)
+        self.assertEqual(ctx.exception.resp.status_code, 400)
+
+    def test_admin_missing_param(self):
+        query = f"FOR vert IN {_VERT_COLL} FILTER vert._key == @id RETURN vert"
+        with self.assertRaises(RERequestError) as ctx:
+            self.client.admin_query(query, bind_vars={})
+        self.assertEqual(ctx.exception.resp.status_code, 400)
+
+    def test_admin_raise_not_found(self):
+        query = f"FOR vert IN {_VERT_COLL} FILTER vert._key == @id RETURN vert"
+        _id = str(uuid4())
+        bind_vars = {'id': _id}
+        with self.assertRaises(RENotFound) as ctx:
+            self.client.admin_query(query, bind_vars, raise_not_found=True)
+        self.assertTrue(_id in ctx.exception.req_body)
+
```

Comment:
A failing auth test is pretty important IMO

Reply:
Applies to the other functions as well for bad tokens. This fn also needs a user vs admin token test